third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -2502,6 +2547,10 @@ static void InitializeRecordReplayApiObjects(v8::Isolate* isolate, LocalFrame* l
   DefineProperty(isolate, context->Global(), "__RECORD_REPLAY_ARGUMENTS__",
                  args);
 
+  v8::Local<v8::Object> globals = v8::Object::New(isolate);
+  DefineProperty(isolate, context->Global(), "__RECORD_REPLAY_GLOBALS__",
```

Comment:
This will reset the globals on every root-level navigation. We probably want to keep them around instead.

Reply:
yeah, I was also confused about this. do we need to store the scripts somewhere so we can rerun them?

Reply:
Domi and I have been chatting about our context handling, which is a tad naiive to say the least; we aren't going to worry about it in this PR, but rather address it wholly in another.